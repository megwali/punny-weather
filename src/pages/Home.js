import { CenteredTextContainer, LinkedCard, LinkedCardContainer } from './styledComponents';

const Home = () => (
  <CenteredTextContainer>
    <h2>Welcome to the punny weather forecast service</h2>
    <h3>How would you like your weather served?</h3>

    <LinkedCardContainer>
      <LinkedCard to="/locate">
        <h3>Misty Magic</h3>
        <p>View weather information about your current location:</p>
        <p>Pick this card and we'll handle the magic of presenting the weather forecast for your location.</p>
        <span>Get my location</span>
      </LinkedCard>

      <LinkedCard to="/search">
        <h3>Sunny Science</h3>
        <p>Search weather forecast information for any city:</p>
        <p>Click this card to be transported to the search portal to find forecast locations and information yourself.</p>
        <span>Take me to the search portal</span>
      </LinkedCard>
    </LinkedCardContainer>
  </CenteredTextContainer>
);

export default Home;
