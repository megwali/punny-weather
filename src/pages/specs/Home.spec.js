import { create } from 'react-test-renderer';
import Home from '../Home';

jest.mock('../../components/styledComponents', () => ({
  CenteredTextContainer: 'CenteredTextContainer',
  LinkedCard: 'LinkedCard',
  LinkedCardContainer: 'LinkedCardContainer',
}));


describe('Home', () => {
  it('renders', () => {
    const component = create(<Home />);
    expect(component.toJSON()).toMatchSnapshot();
  });
})
