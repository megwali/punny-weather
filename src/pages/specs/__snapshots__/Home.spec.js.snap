// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<CenteredTextContainer>
  <h2>
    Welcome to the punny weather forecast service
  </h2>
  <h3>
    How would you like your weather served?
  </h3>
  <LinkedCardContainer>
    <LinkedCard
      to="/locate"
    >
      <h3>
        Misty Magic
      </h3>
      <p>
        View weather information about your current location:
      </p>
      <p>
        Click here and we'll handle the magic of presenting the weather forecast for your location.
      </p>
      <Link
        to="/locate"
      >
        Get my location
      </Link>
    </LinkedCard>
    <LinkedCard
      to="/search"
    >
      <h3>
        Sunny Science
      </h3>
      <p>
        Search weather forecast information for any city:
      </p>
      <p>
        Click here to be transported to the search portal to find forecast locations and information yourself.
      </p>
      <Link
        to="/search"
      >
        Take me to the search portal
      </Link>
    </LinkedCard>
  </LinkedCardContainer>
</CenteredTextContainer>
`;
